### PR TITLE
ETT article choose a library or framework

### LINKED ISSUE #7842: Choose a JavaScript library or framework [6/5/2022]
* Proposed title: Choose a JavaScript library or framework
* Proposed by: johanneq
* Related areas: None of the above
* Content type: Explanation (https://documentation.divio.com/explanation/)
* Deadline: 6/5/2022
* Deadline explanation: One month will be fine - the pathways on DGC where this content belongs will be live for ETT prior to this date, but they will not need to review right away.
* [Draft](https://docs.google.com/document/d/1PbohHIT_h9waLXwER8LT7gEnsrcshUJze46J8MY4uTg/edit?resourcekey=0-WReHwS8A7QWJmycgJbPccA#)
* Content summary: This article is part of the Web Fundamentals Course launching to the public in September, and to the ETT program in April. This article shares insights on how you might pick a library or framework to use within your web application.

### PR DESCRIPTION
**Pull request by rachelandrew**
Fixes #7842 

Preview: https://deploy-preview-7977--web-dev-staging.netlify.app/choose-js-library-or-framework/

### CODE REVIEW

**Review by umaar — COMMENTED** [review]
A few small suggestions!

File: src/site/content/en/web-dev-basics-one/choose-js-library-or-framework/index.md
Comment (malchata) [inline src/site/content/en/web-dev-basics-one/choose-js-library-or-framework/index.md:38]:
```suggestion
The following example shows a code snippet that uses the [Preact](https://preactjs.com/) JavaScript framework:
```

File: src/site/content/en/web-dev-basics-one/choose-js-library-or-framework/index.md
Comment (malchata) [inline src/site/content/en/web-dev-basics-one/choose-js-library-or-framework/index.md:56]:
```suggestion
Some client-side JavaScript libraries help abstract away quirks of the web platform. A library can also serve as a learning tool. For example, if you're unfamiliar with [animation easing functions](https://easings.net/), the source code of a library can teach you how such easings work.
```

File: src/site/content/en/web-dev-basics-one/choose-js-library-or-framework/index.md
Comment (malchata) [inline src/site/content/en/web-dev-basics-one/choose-js-library-or-framework/index.md:70]:
```suggestion
Technically, you can develop your web application from scratch, but why go to the trouble when you can use free (open source) software, or purchase a solution that, in the long run, can save time and money? There are a large number of JavaScript libraries and frameworks available, each offering a unique approach to solving problems, and each having different characteristics. For example, a library can:

-  Be written and maintained internally rather than by a third party.
-  Have specific legal licenses that make it suitable or unsuitable for your web application.
-  Be outdated or unmaintained.
-  Simplify a set of complex tasks and save you a lot of time and money.
-  Be widely used in the community, and can be well known amongst developers.
```

File: src/site/content/en/web-dev-basics-one/choose-js-library-or-framework/index.md
Comment (malchata) [inline src/site/content/en/web-dev-basics-one/choose-js-library-or-framework/index.md:74]:
```suggestion
{% Aside 'important' %}
```

Comment (malchata) [reply src/site/content/en/web-dev-basics-one/choose-js-library-or-framework/index.md:74]: I feel this rises to the level of an 'important' aside, since it's a key performance consideration.

File: src/site/content/en/web-dev-basics-one/choose-js-library-or-framework/index.md
Comment (malchata) [inline src/site/content/en/web-dev-basics-one/choose-js-library-or-framework/index.md:86]: No correction here, just a stellar callout that needed to be said.

File: src/site/content/en/web-dev-basics-one/choose-js-library-or-framework/index.md
Comment (malchata) [inline src/site/content/en/web-dev-basics-one/choose-js-library-or-framework/index.md:109]:
```suggestion
-  If you use GitHub, then consider the code's security [offerings](https://github.com/features/security), such as [Dependabot](https://docs.github.com/en/code-security/supply-chain-security/keeping-your-dependencies-updated-automatically/about-dependabot-version-updates). Or, consider alternative services that scan for vulnerabilities in your code, such as [snyk.io](https://snyk.io/).
```

File: src/site/content/en/web-dev-basics-one/choose-js-library-or-framework/index.md
Comment (malchata) [inline src/site/content/en/web-dev-basics-one/choose-js-library-or-framework/index.md:166]:
```suggestion
{% Aside 'warning' %}
This is not legal advice.
{% endAside %}
```

**Review by malchata — APPROVED** [review]
Other than minor nits, LGTM!

File: src/site/content/en/web-dev-basics-one/choose-js-library-or-framework/index.md
Comment (umaar) [reply src/site/content/en/web-dev-basics-one/choose-js-library-or-framework/index.md:70]: fyi the reason for the extra clarity/redundancy in my bullet points is to a) generally be more explicit/clear to the reader, b) maybe avoid stuff like https://twitter.com/soft/status/1449406390976409600 !

I don't mind if it needs to be changed though 👍

Comment (malchata) [reply src/site/content/en/web-dev-basics-one/choose-js-library-or-framework/index.md:70]: Your call! I felt it flowed better this way, but your way is fine too! :)

**Comment by malchata** [comment]
@rachelandrew, are we merging these as soon as we get approval, or waiting?

File: src/site/content/en/web-dev-basics-one/choose-js-library-or-framework/index.md
Comment (rachelandrew) [reply src/site/content/en/web-dev-basics-one/choose-js-library-or-framework/index.md:70]: that tweet is interesting, probably something for writers to discuss, but I'll leave this as is.

**Comment by umaar** [comment]
LGTM!

(don't have GitHub permission on this repo to select `Review Changes > Approve`)

**Pull request merged**
